src/local-mode.ts

```diff
@@ -56,42 +90,172 @@ export class LocalModeManager {
 
 		if (!localMode) {
 			// Remove any leftovers if necessary
-			await this.removeLocalModeArtifacts();
+			await this.handleLocalModeStateChange(false);
 		}
 	}
 
-	public async removeLocalModeArtifacts(): Promise<void> {
-		try {
-			const images = await this.getLocalModeImages();
-			const containers = await this.getLocalModeContainers(images);
+	// Query the engine to get currently running containers and installed images.
+	public async collectEngineSnapshot(): Promise<EngineSnapshotRecord> {
+		const containers = this.docker
+			.listContainers()
+			.then(resp => _.map(resp, 'Id'));
+		const images = this.docker.listImages().then(resp => _.map(resp, 'Id'));
+		const volumes = this.docker
+			.listVolumes()
+			.then(resp => _.map(resp.Volumes, 'Name'));
+		const networks = this.docker.listNetworks().then(resp => _.map(resp, 'Id'));
+
+		return Bluebird.all([containers, images, volumes, networks]).then(
+			data =>
+				new EngineSnapshotRecord(
+					new EngineSnapshot(data[0], data[1], data[2], data[3]),
+					new Date(),
+				),
+		);
+	}
+
+	// Store engine snapshot data in the local database.
+	public async storeEngineSnapshot(
+		record: EngineSnapshotRecord,
+	): Promise<void> {
+		const timestamp = record.timestamp.toISOString();
+		log.info(
+			`Storing engine snapshot in the database. Timestamp: ${timestamp}`,
+		);
+		return this.db
+			.models('engineSnapshot')
+			.delete()
+			.then(() => {
+				return this.db.models('engineSnapshot').insert({
+					snapshot: JSON.stringify(record.snapshot),
+					timestamp,
+				});
+			});
```

Comment:
Please use `await` here (or drop the `async` qualifier in the function definition, but I'd prefer you use `await`).